test/e2e/provisioning/aws_e2e_test.go

```diff
@@ -0,0 +1,102 @@
+// +build e2e
+
+package provisioning
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+)
+
+const (
+	aws_manifest = "./testdata/machine-aws.yaml"
+)
+
+var awsScenarios = []struct {
+	// name holds short description of the test scenario, it is also used to create machines and nodes names
+	// so please don't put "strange" characters there
+	name                    string
+	osName                  string
+	containerRuntime        string
+	containerRuntimeVersion string
+	// if the -short flag was provided and this variable is set
+	// the test scenario will be run otherwise it will be skipped.
+	short bool
+}{
+	{
```

Comment:
Wouldn'it it make more sense to have a `Scenerios` struct that accepts a `providerConfig` and can be re-used across different providers?

Reply:
Escpecially since we most likely want to run the same set of tests for each provider

Reply:
I think that table driven test are quite elegant. Provider specific tests only prepare input for the `verify` tool, the rest of the logic is shared across providers.

Reply:
I am not sure I understand your comment. We have the same table of tests for all providers, which is also reflected in the code here, as there is `awsScenarios ` and `digitalOceanScenarios`, both with the same content - I think it makes more sense to have a global struct `scenarios`  (or `e2eScenenarios` maybe) which is imported and used by all provider

Reply:
ok, I'll create `scenarios` struct and place it it `helpers.go` file.